ls1intum/Artemis
# Bugfix/course/archive/various fixes

### Checklist

### Motivation and Context


The current implementation of the course archive has some issues:
- Various temporary directories weren't deleted after they were no longer needed
- An exception was thrown when a programming exercise did not have any repository URLs
- Unnecessary logging polluted the logs
- The date format used throughout the zip files included slashes which can be problematic
-  The general structure of the zip files wasn't optimal

### Description

This PR resolved the issues mentioned above.

### Steps for Testing


1. Log in to Artemis
2. Navigate to Course Administration
3. Generate some exercises (supported are: programming, text, file upload, and modeling)
4. Participate in these exercises with various users. The aim is to trigger some submissions for theses exercises
5. Set the course end date to the past so that it's over 
6. Navigate to the course detail screen and press the Archive Course button
7. Wait until archiving is complete and download the archive
8. The archive name should be {Course-Short-Name}-{Course-Title}-{Timestamp}.zip
9. The zipped files inside the archive should be {Course-Short-Name}-{Exercise-Title}-{Timestamp}.zip
10. Also make sure that the student submissions are included in the zip files. Programming exercises also have the template, solution, and test repositories

### Test Coverage



SubmissionExportService 91%
FileService 83%
CourseExportService 67%
ProgrammingExerciseExportService 42%
GitService 74%
### Screenshots
